packages/config/src/VitessceAutoConfig.js

```diff
@@ -0,0 +1,380 @@
+
+    if (hasCellSetData.length > 0) {
+      views.push(['obsSets']);
+    }
+
+    this.metadataSummary.obsm.forEach((key) => {
+      if (key.toLowerCase().includes('obsm/x_umap')) {
+        views.push(['scatterplot', { mapping: 'UMAP' }]);
+      }
+      if (key.toLowerCase().includes('obsm/x_tsne')) {
+        views.push(['scatterplot', { mapping: 't-SNE' }]);
+      }
+      if (key.toLowerCase().includes('obsm/x_pca')) {
+        views.push(['scatterplot', { mapping: 'PCA' }]);
+      }
+      if (key.toLowerCase().includes(('obsm/x_segmentations'))) {
+        views.push(['layerController']);
+      }
+      if (key.toLowerCase().includes(('obsm/x_spatial'))) {
+        views.push(['spatial']);
+      }
+    });
+
+    if (this.metadataSummary.X) {
+      views.push(['heatmap']);
+      views.push(['featureList']);
+    }
+
+    console.log("HERE NOW ", views);
+
+    return views;
+  }
+
+  async setMetadataSummary() {
+    if (Object.keys(this.metadataSummary).length > 0) {
+      return this.metadataSummary;
+    }
+
+    const parseMetadataFile = (metadataFile) => {
+
+      if (!metadataFile.metadata) {
+        throw new Error('Could not generate config: .zmetadata file is not valid.');
+      }
+
+      const obsmKeys = Object.keys(metadataFile.metadata)
+        .filter(key => key.startsWith('obsm/X_'))
+        .map(key => key.split('/.zarray')[0]);
+
+      const obsKeysArr = Object.keys(metadataFile.metadata)
+        .filter(key => key.startsWith('obs/') && !key.includes('obs/.') && !key.includes('obs/__'))
+        .map(key => key.split('/.za')[0]);
+
+      function uniq(a) {
+        return a.sort().filter((item, pos, ary) => !pos || item !== ary[pos - 1]);
+      }
+      const obsKeys = uniq(obsKeysArr);
+
+      const X = Object.keys(metadataFile.metadata).filter(key => key.startsWith('X'));
+
+      const out = {
+        obsm: obsmKeys,
+        obs: obsKeys,
+        X: X.length > 0,
+      };
+
+      return out;
+    };
+
+    const metadataExtension = '.zmetadata';
+    const url = [this.fileUrl, metadataExtension].join('/');
+    return fetch(url).then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      return Promise.reject(response);
+    })
+      .then(responseJson => parseMetadataFile(responseJson))
+      .catch((error) => {
+        if (error.status === 404) {
+          const errorMssg = `Could not generate config. File ${metadataExtension} not found in supplied file URL. Check docs for more explanation.`;
+          return Promise.reject(new Error(errorMssg));
+        }
+        return Promise.reject(error);
+      });
+  }
+}
+
+const configClasses = [
+  {
+    extensions: ['ome.tif', '.ome.tiff', '.ome.tf2', '.ome.tf8'], // todo: test that ".ome.tf2", ".ome.tf8" work
+    class: OmeTiffAutoConfig,
+  },
+  {
+    extensions: ['h5ad.zarr', '.adata.zarr', '.anndata.zarr'],
```

Comment:
FWIW this is extremely nonstandard.  I am not sure how to handle this, but I personally have been saving anndata stores with just the `zarr` extension.  Maybe you could comment here @keller-mark as to what you think we should handle vs. not.

Reply:
I suggested these file extensions to @ivababukova.

I don't think many people currently save their AnnData as Zarr stores in the first place, so I figured we could start to recommend this in our docs - maybe it will catch on.

I have personally found it helpful and have been using it in vitessce-python for demo data processing. For example, see this directory where there are both OME and AnnData Zarr stores for the same dataset: https://github.com/vitessce/vitessce-python/blob/main/demos/human-lymph-node-10x-visium/config.yml

Reply:
There is a lot of room of ambiguity with these extensions -- the situation is the same with OME-ZARR files too. I feel that the only realistic way to work with that is to make some assumptions (like, this file format is of type X if it has those extensions) and make sure our assumptions are super clear and well documented. If there are some other assumptions that are more practical to make, please let me know @ilan-gold .